docs/src/internals.md

```diff
@@ -485,20 +502,158 @@ julia> coefnames(f.rhs)
+StatsModels.TableRegressionModel{LinearModel{GLM.LmResp{Array{Float64,1}},GLM.DensePredChol{Float64,LinearAlgebra.Cholesky{Float64,Array{Float64,2}}}},Array{Float64,2}}
+
+y ~ 1 + poly(a, 2) + poly(b, 2)
+
+Coefficients:
+────────────────────────────────────────────────────────────────────────────
+              Estimate  Std. Error   t value  Pr(>|t|)  Lower 95%  Upper 95%
+────────────────────────────────────────────────────────────────────────────
+(Intercept)   0.693521   0.159469    4.34893    <1e-4    0.376934    1.01011
+a^1           2.1042     0.383115    5.49235    <1e-6    1.34362     2.86478
+a^2           2.34395    1.39244     1.68334    0.0956  -0.420386    5.10829
+b^1          -0.180113   0.148698   -1.21127    0.2288  -0.475317    0.11509
+b^2           2.89786    0.0794572  36.4707     <1e-56   2.74012     3.05561
+────────────────────────────────────────────────────────────────────────────
+```
+
+### [Making special syntax "runtime friendly"] (@id extend-runtime)
+
+When used from the `@formula` macro, special syntax relies on dispatching on the
+`FunctionTerm{MyFunction}` type.  But when creating a formula at runtime
+without the `@formula` macro, `FunctionTerm`s aren't available, and so care must
+be taken to make sure you provide a runtime replacement.  The example for `poly`
+above shows how to do this, but we spell it out here in more detail.
+
+The first step is to make sure you can create a schema-less instance of the
+`AbstractTerm` that implements your special syntax behavior.  That's why for the
+`poly` example, that means we need to be able to create a
+`PolyTerm(term(column_name), term(poly_degree))`.  In order to do this, the
+types of the `term` and `deg` fields aren't specified but are parameters of the
+`PolyTerm` type.
+
+The second step is to provide an `apply_schema` method that upgrades a
+schema-less instance to one with a schema (i.e., one that can be used with
+`modelcols`).  For example, we've specified `apply_schema(pt::PolyTerm, ...)`
+which calls `apply_schema` on the wrapped `pt.term`, returning a new `PolyTerm`
+with the instantiated result.  This might seem like unnecessary indirection,
+given that `apply_schema(::FunctionTerm{typeof(poly)}, ...)` simply returns a
+schema-less `PolyTerm`.  But it means that users can do things like
+
+```jldoctest 1
+julia> ff = apply_schema(term(:y) ~ term(1) + term(:a) + PolyTerm(term(:b), term(2)),
+                         schema(data),
+                         StatisticalModel)
+FormulaTerm
+Response:
+  y(continuous)
+Predictors:
+  1
+  a(continuous)
+  poly(b, 2)
+
+julia> modelcols(ff.rhs, data)
+4×4 Array{Float64,2}:
+ 1.0  0.488613  1.0   1.0
+ 1.0  0.210968  2.0   4.0
+ 1.0  0.951916  3.0   9.0
+ 1.0  0.999905  4.0  16.0
+```
+
+This is workable if a little verbose, so the third (optional) step is to provide
+additional "runtime" methods for the function that's used in the `@formula`
```

Comment:
```suggestion
additional "runtime" methods for the function that are used in the `@formula`
```

Reply:
"function" is the head of this relative clause so single number is appropriate no?  That is, it's the function that's used, not the methods...although now I see that this is potentially structurally ambiguous :)